src/main/scala/org/broadinstitute/hail/io/IndexedBinaryBlockReader.scala

```diff
@@ -5,27 +5,21 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 import org.apache.hadoop.io.LongWritable
 import org.apache.hadoop.mapred._
-import org.broadinstitute.hail.variant.Genotype
-import org.broadinstitute.hail.annotations._
+
 import scala.collection.mutable
 
-class VariantRecord[K] extends Serializable {
-  var gs: Iterable[Genotype] = null
-  var ann: Annotation = Annotation.empty
+abstract class BinaryDecoder[K, V] extends Serializable {
```

Comment:
I think this name can be more evocative.  I figured out what it does from the code.  It seems it is a key value record where the value is decoded from an Array on demand, yes?  How about `KeySerializedValueRecord`?  Then I might rename `setInput` to `setSerializedValue` and drop `decode` and just make `getValue` abstract.

Thoughts?